test/unit/deadline_client/cli/test_cli_asset.py

```diff
@@ -155,3 +204,212 @@ def test_asset_snapshot_recursive(
         actual_inputs = set(mock_prepare_paths_for_upload.call_args[1]["input_paths"])
         assert actual_inputs == expected_inputs
         mock_hash_attachments.assert_called_once()
+
+
+class TestUpload:
+
+    def test_upload_valid_manifest(
+        fresh_deadline_config, mock_init_objects, mock_upload_attachments
+    ):
+        """
+        Test the asset upload command with correct arguments and valid manifest path.
+        """
+        with patch.object(_submit_job_bundle.api, "get_boto3_client"), patch.object(
+            _submit_job_bundle.api, "get_queue_user_boto3_session"
+        ), patch.object(os.path, "isdir", side_effect=[True, True]), patch.object(
+            os, "listdir", return_value=["manifest_input"]
+        ), patch.object(
+            asset_group, "read_local_manifest", return_value=mock_asset_manifest
+        ), patch.object(
+            asset_group, "diff_manifest", return_value=[]
+        ), patch.object(
+            S3AssetUploader, "_write_local_manifest_s3_mapping"
+        ) as mock_write_manifest_mapping:
+
+            runner = CliRunner()
+            result = runner.invoke(
+                main,
+                [
+                    "asset",
+                    "upload",
+                    "--root-dir",
+                    MOCK_ROOT_DIR,
+                    "--manifest-dir",
+                    MOCK_MANIFEST_DIR,
+                    "--farm-id",
+                    MOCK_FARM_ID,
+                    "--queue-id",
+                    MOCK_QUEUE_ID,
+                ],
+            )
+
+            full_manifest_key = MOCK_UPLOAD_ATTACHMENTS_RESPONSE["manifests"][0][
+                "inputManifestPath"
+            ]
+            manifest_name = os.path.basename(full_manifest_key)
+            manifest_dir_name = os.path.basename(MOCK_MANIFEST_DIR)
+
+            mock_write_manifest_mapping.assert_called_once_with(
+                manifest_write_dir=MOCK_ROOT_DIR,
+                manifest_name=manifest_name,
+                full_manifest_key=full_manifest_key,
+                manifest_dir_name=manifest_dir_name,
+            )
+            mock_upload_attachments.assert_called_once()
+            assert result.exit_code == 0
```

Comment:
nit: We don't need this assertion code inside the `with` block

Same for other assertion code below that is inside the `with` block unnecessarily